src/main/webapp/src/App.js

```diff
@@ -10,35 +10,52 @@ import Phones from "./components/Phones";
 import MainPage from "./components/MainPage";
 import PhoneDetails from "./components/PhoneDetails";
 import NotFoundPage from './components/NotFoundPage';
+import Basket from './components/Basket'
+import EmptyBasketPage from "./components/EmptyBasketPage";
 import {Redirect} from "react-router";
 
 const {Header, Footer, Content} = Layout;
 
 
-function App() {
-    return (
-
-            <Layout className="mainLayout">
-                <Header>
-                    <AppHeader/>
-                </Header>
-                <Content>
-                    <Switch>
-                    <Route exact path="/" component={MainPage} />
-                    <Route exact path="/phones" component={Phones}/>
-                    <Route exact path="/phones/:id" component={PhoneDetails} />
-                    <Route exact path="/404" component={NotFoundPage} />
-                    <Route>
-                        <Redirect to="/404"/>
-                    </Route>
-                    </Switch>
-                </Content>
-                <Footer>
-                    <AppFooter/>
-                </Footer>
-            </Layout>
-
-    );
+class App extends React.Component{
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            basket: null,
+        }
+    }
+
+
+ render() {
+
+     return (
+         <Layout className="mainLayout">
+             <Header>
+                 <AppHeader/>
+             </Header>
+             <Content>
+                 <Switch>
+                     <Route exact path="/" component={MainPage} />
+                     <Route exact path="/phones" component={Phones}/>
+                     <Route exact path="/phones/:id" component={PhoneDetails} />
+                     <Route exact path="/404" component={NotFoundPage} />
+                     <Route exact path="/basket" component={Basket} />
+                     <Route exact path="/emptybasket" component={EmptyBasketPage} />
```

Comment:
make one single component with one single url to show either empty or non-empty basket

Reply:
+